lib/mix/test/mix/tasks/compile.elixir_test.exs

```diff
@@ -438,4 +438,50 @@ defmodule Mix.Tasks.Compile.ElixirTest do
       end)
     end
   end
+
+  test "does not use incorrect line number when error originates in another file" do
```

Comment:
Can you please move this to the parallel_compiler_test.exs? I don't think we should test it through mix. You can use `Path.join(tmp_path(), "parallel_compiler_undef.exs")` in there to generate a path in the temp directory where you will write the file.